### PR TITLE
minizip: new recipe

### PR DESCRIPTION
**Pull request by miqlas**

### CODE REVIEW

File: app-arch/minizip/minizip-1.2.11.recipe [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+SOURCE_URI="http://zlib.net/fossils/zlib-$portVersion.tar.gz"
+CHECKSUM_SHA256="c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1"
+SOURCE_DIR="zlib-$portVersion"
+PATCHES="minizip-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	minizip$secondaryArchSuffix = $portVersion compat >= 1
```
Comment (korli) [inline app-arch/minizip/minizip-1.2.11.recipe]: you can drop the compat on the package.

File: app-arch/minizip/minizip-1.2.11.recipe [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	lib:libminizip$secondaryArchSuffix = $portVersion
```
Comment (korli) [inline app-arch/minizip/minizip-1.2.11.recipe]: The version isn't 1.0.0?

**Comment by korli** [comment]
minizip could theorically be a subpackage of zlib.

File: app-arch/minizip/minizip-1.2.11.recipe (hunk 2 above)
Comment (korli) [reply app-arch/minizip/minizip-1.2.11.recipe]: This isn't taken care of?

**Comment by diversys** [comment]
This one is needed for Telegram.

**Comment by Neustradamus** [comment]
Zlib PR 830 from @gvollant Minizip author:
- https://github.com/madler/zlib/pull/830

"Support Haiku in minizip." merged commit in upstream Zlib develop branch:
- https://github.com/madler/zlib/commit/aa154e3da0d98629bbefe79cdfba8bc7178dc4ad

There is a little change.

Original HaikuPorts PR from Zoltán Mizsei @extrowerk (merged by @diversys) on GitHub:
- https://github.com/haikuports/haikuports/pull/1593

Zlib and Minizip HaikuPorts files:
- https://github.com/haikuports/haikuports/commits/master/app-arch/minizip
- https://github.com/haikuports/haikuports/commits/master/sys-libs/zlib/patches

Updated HaikuPorts PR from @Begasus:
- https://github.com/haikuports/haikuports/pull/7394

Requested PR in Zlib upstream from @Begasus:
- https://github.com/madler/zlib/pull/741

cc: @madler, @waddlesplash, @korli, @pulkomandy, @fbrosson.

**Pull request merged**